### PR TITLE
Feature/rb 40 scene manager class

### PR DESCRIPTION
**Pull request by guillaumeAbel**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/Client/Systems/ClientSystems.cpp
Comment (guillaumeAbel) [inline src/Client/Systems/ClientSystems.cpp:1]: fichier séparer en dossiers Events/ et Graphic/

File: src/Client/Systems/Graphic/GraphicSystems.cpp
Comment (guillaumeAbel) [inline src/Client/Systems/Graphic/GraphicSystems.cpp:1]: des qu'on devra rajouter un system graphic on separera ce fichier en Sprite, Text, Audio etc psk il est deja bien gros

File: src/Client/SceneManager.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+{
+}
+
+static void initRaylib()
+{
+    InitWindow(
+        screenWidth,
+        screenHeight,
+        "raylib [textures] examples - texture source and destination "
+        "rectangles");
```
Comment (Saverio976) [inline src/Client/SceneManager.cpp]:
```suggestion
        "R-Bus");
```

en vrai ça serait bien de commencer à mettr ele nom du jeu

File: src/main_client.cpp [hunk 2]
```diff
@@ -5,6 +5,9 @@
 ** main
 */
 
+#include "SceneManager.hpp"
+
 int main()
 {
+    SceneManager::getInstance();
```
Comment (Saverio976) [inline src/main_client.cpp]: je suis pas particulièrement réjoui par ça

ça serait cool de faire un truc du style

```c
game = SceneManager::getInstance();

errors = game.run();

return errors;
```

File: src/Client/SceneManager.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+SceneManager &SceneManager::getInstance()
+{
+    if (!_init) {
+        _init = true;
+        _instance.init();
+    }
+    return _instance;
+}
+
+SceneManager::SceneManager() : _currentScene(Scene::MAIN_GAME), _stop(false)
```
Comment (Saverio976) [inline src/Client/SceneManager.cpp:49]: _currentScene c'est utilisé où dans ce code ?

**Pull request merged**